File: numberCombination/numberCombination.java
package numberCombination;

public class numberCombination {

    public static void generateCombinations(int a, int b, int c, String current) {
        if (current.length() == 3) {
            System.out.print(current + " ");
            return;
        }

        generateCombinations(a, b, c, current + a);
        generateCombinations(a, b, c, current + b);
        generateCombinations(a, b, c, current + c);
    }

    public static void main(String[] args) {
        int a = 1;
        int b = 2;
        int c = 3;

        generateCombinations(a, b, c, "");

    }

}

File: numberCombination/numberCombination.js
function generateCombinations(a, b, c, current, combinations) {
  if (current.length === 3) {
    combinations.push(current);
    return;
  }

  generateCombinations(a, b, c, current + a, combinations);
  generateCombinations(a, b, c, current + b, combinations);
  generateCombinations(a, b, c, current + c, combinations);
}

const a = 1;
const b = 2;
const c = 3;
const combinations = [];

generateCombinations(a, b, c, "", combinations);

console.log(combinations);
